From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Fri, 17 Mar 2023 20:01:40 +0100
Subject: media: Use gstplayer from meson subproject

Origin: https://gitlab.gnome.org/3v1n0/gtk/-/commits/gst-player-subproject
Forwarded: not-needed
---
 modules/media/meson.build | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/modules/media/meson.build b/modules/media/meson.build
index 74364e6..556098f 100644
--- a/modules/media/meson.build
+++ b/modules/media/meson.build
@@ -39,6 +39,9 @@ if ffmpeg_found
   )
 endif
 
+subproject('gstreamer-player-1.0',
+           required: get_option('media-gstreamer'),
+           default_options: 'default_library=static')
 gstplayer_dep = dependency('gstreamer-player-1.0', version: '>= 1.12.3',
                            required: get_option('media-gstreamer'))
 gstgl_dep = dependency('gstreamer-gl-1.0', version: '>= 1.12.3',
