From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Fri, 17 Mar 2023 20:01:40 +0100
Subject: media: Use gstplayer from meson subproject

Origin: https://gitlab.gnome.org/3v1n0/gtk/-/commits/gst-player-subproject
Forwarded: not-needed
---
 meson.build | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/meson.build b/meson.build
index a450e47..0fc1d1c 100644
--- a/meson.build
+++ b/meson.build
@@ -411,6 +411,10 @@ if win32_enabled
   pangowin32_dep = dependency('pangowin32')
 endif
 
+subproject('gstreamer-player-1.0',
+           required: get_option('media-gstreamer'),
+           default_options: 'default_library=static')
+
 pangocairo_dep    = dependency('pangocairo', version: pango_req)
 pixbuf_dep        = dependency('gdk-pixbuf-2.0', version: gdk_pixbuf_req,
                                default_options: ['png=enabled', 'jpeg=enabled', 'builtin_loaders=png,jpeg', 'man=false'])
