From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Wed, 28 Jan 2026 14:36:19 +0100
Subject: tests: Mark memorytexture as expected to fail on big-endian machines

Forwarded: not-needed, workaround
---
 testsuite/gdk/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/testsuite/gdk/meson.build b/testsuite/gdk/meson.build
index f6fcc46..45af858 100644
--- a/testsuite/gdk/meson.build
+++ b/testsuite/gdk/meson.build
@@ -19,7 +19,7 @@ tests = [
   { 'name': 'displaymanager' },
   { 'name': 'glcontext' },
   { 'name': 'keysyms' },
-  { 'name': 'memorytexture' },
+  { 'name': 'memorytexture', 'suites': host_machine.endian() == 'big' ? ['failing'] : [] },
   { 'name': 'rectangle' },
   { 'name': 'rgba' },
   { 'name': 'seat' },
