From: Simon McVittie <smcv@debian.org>
Date: Tue, 22 Aug 2023 11:57:36 +0100
Subject: tests: Mark gltexture as expected to fail on big-endian machines

Forwarded: not-needed, workaround
---
 testsuite/gdk/meson.build | 4 ++++
 1 file changed, 4 insertions(+)

Index: gtk4-4.14.1+ds/testsuite/gdk/meson.build
===================================================================
--- gtk4-4.14.1+ds.orig/testsuite/gdk/meson.build	2024-03-20 17:00:36.058661277 +0100
+++ gtk4-4.14.1+ds/testsuite/gdk/meson.build	2024-03-20 17:00:36.054661336 +0100
@@ -85,6 +85,10 @@
 
   suites = ['gdk'] + t.get('suites', [])
 
+  if host_machine.endian() == 'big' and t.get('name') == 'gltexture'
+    suites += 'failing'
+  endif
+
   test(test_name, test_exe,
     args: [ '--tap', '-k' ],
     protocol: 'tap',
