From: Sebastien Bacher <seb128@ubuntu.com>
Date: Fri, 5 Apr 2024 16:09:34 +0200
Subject: tests: Mark memorytexture as expected to fail on big-endian machines

Forwarded: not-needed, workaround
---
 testsuite/gdk/meson.build | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/testsuite/gdk/meson.build b/testsuite/gdk/meson.build
index 6917e0b..2bb83fa 100644
--- a/testsuite/gdk/meson.build
+++ b/testsuite/gdk/meson.build
@@ -48,6 +48,10 @@ foreach t : tests
 
   suites = ['gdk'] + t.get('suites', [])
 
+  if host_machine.endian() == 'big' and t.get('name') == 'memorytexture'
+    suites += 'failing'
+  endif
+
   test(test_name, test_exe,
     args: [ '--tap', '-k' ],
     protocol: 'tap',
@@ -111,6 +115,10 @@ if get_option('install-tests')
     test_name = t.get('name')
     suites = t.get('suites', [])
 
+    if host_machine.endian() == 'big' and t.get('name') == 'memorytexture'
+      suites += 'failing'
+    endif
+
     if suites.contains('flaky') or suites.contains('failing')
       continue
     endif
