From: Simon McVittie <smcv@debian.org>
Date: Tue, 22 Aug 2023 11:57:36 +0100
Subject: tests: Mark gltexture as expected to fail on big-endian machines

Forwarded: not-needed, workaround
---
 testsuite/gdk/meson.build | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/testsuite/gdk/meson.build b/testsuite/gdk/meson.build
index 2ae1847..4138f86 100644
--- a/testsuite/gdk/meson.build
+++ b/testsuite/gdk/meson.build
@@ -101,6 +101,10 @@ foreach t : internal_tests
 
   suites = ['gdk'] + t.get('suites', [])
 
+  if host_machine.endian() == 'big' and t.get('name') == 'gltexture'
+    suites += 'failing'
+  endif
+
   test(test_name, test_exe,
     args: [ '--tap', '-k' ],
     protocol: 'tap',
