From: Alessandro Astone <alessandro.astone@canonical.com>
Date: Tue, 10 Feb 2026 17:04:22 +0100
Subject: testsuite: Skip vulkan tests that break lavapipe

Forwarded: not-needed, workaround
---
 testsuite/gdk/memorytexture.c | 15 +++++++++++++++
 testsuite/gsk/scaling.c       | 36 ++++++++++++++++++++++++++++++++++++
 2 files changed, 51 insertions(+)

diff --git a/testsuite/gdk/memorytexture.c b/testsuite/gdk/memorytexture.c
index 00d9887..d6430fd 100644
--- a/testsuite/gdk/memorytexture.c
+++ b/testsuite/gdk/memorytexture.c
@@ -451,6 +451,15 @@ should_skip_download_test (GdkMemoryFormat format,
           g_test_skip ("Vulkan is not supported");
           return TRUE;
         }
+      else if (format == GDK_MEMORY_G8_B8R8_444 ||
+               format == GDK_MEMORY_G8_R8B8_444 ||
+               format == GDK_MEMORY_G8_B8_R8_444 ||
+               format == GDK_MEMORY_G8_R8_B8_444 ||
+               format == GDK_MEMORY_G16_B16_R16_444)
+        {
+          g_test_skip ("format is broken in lavapipe");
+          return TRUE;
+        }
       return FALSE;
 
     case TEXTURE_METHOD_PNG_PIXBUF:
@@ -490,6 +499,12 @@ test_download (gconstpointer data,
   if (should_skip_download_test (format, method))
     return;
 
+  if (method == TEXTURE_METHOD_VULKAN && width > 1 && height > 1)
+    {
+      g_test_skip ("test triggers SIGBUS in lavapipe, depending on random WxH");
+      return;
+    }
+
   width = round_up (width, gdk_memory_format_get_block_width (format));
   height = round_up (height, gdk_memory_format_get_block_height (format));
 
diff --git a/testsuite/gsk/scaling.c b/testsuite/gsk/scaling.c
index 53902ab..416e594 100644
--- a/testsuite/gsk/scaling.c
+++ b/testsuite/gsk/scaling.c
@@ -312,6 +312,27 @@ test_linear_filtering (gconstpointer data,
       return;
     }
 
+  if (GSK_IS_VULKAN_RENDERER (renderer) &&
+      (format == GDK_MEMORY_G8_B8R8_444 ||
+       format == GDK_MEMORY_G8_R8B8_444 ||
+       format == GDK_MEMORY_G8_B8_R8_444 ||
+       format == GDK_MEMORY_G8_R8_B8_444 ||
+       format == GDK_MEMORY_G16_B16_R16_444 ||
+       format == GDK_MEMORY_G8B8G8R8_422 ||
+       format == GDK_MEMORY_G8R8G8B8_422 ||
+       format == GDK_MEMORY_R8G8B8G8_422 ||
+       format == GDK_MEMORY_B8G8R8G8_422))
+    {
+      g_test_skip ("format is broken in lavapipe");
+      return;
+    }
+
+  if (GSK_IS_VULKAN_RENDERER (renderer) && width == 512)
+    {
+      g_test_skip ("512x512 test triggers SIGBUS in lavapipe");
+      return;
+    }
+
   width = MAX (width, gdk_memory_format_get_block_width (format));
   height = MAX (height, gdk_memory_format_get_block_height (format));
 
@@ -354,6 +375,21 @@ test_mipmaps (gconstpointer data)
       return;
     }
 
+  if (GSK_IS_VULKAN_RENDERER (renderer) &&
+      (format == GDK_MEMORY_G8_B8R8_444 ||
+       format == GDK_MEMORY_G8_R8B8_444 ||
+       format == GDK_MEMORY_G8_B8_R8_444 ||
+       format == GDK_MEMORY_G8_R8_B8_444 ||
+       format == GDK_MEMORY_G16_B16_R16_444 ||
+       format == GDK_MEMORY_G8B8G8R8_422 ||
+       format == GDK_MEMORY_G8R8G8B8_422 ||
+       format == GDK_MEMORY_R8G8B8G8_422 ||
+       format == GDK_MEMORY_B8G8R8G8_422))
+    {
+      g_test_skip ("format is broken in lavapipe");
+      return;
+    }
+
   width = MAX (2, gdk_memory_format_get_block_width (format));
   height = MAX (2, gdk_memory_format_get_block_height (format));
 
