From c44288c7394dc6324a273d248093a311ee6b42b1 Mon Sep 17 00:00:00 2001
From: Sebastien Bacher <seb128@ubuntu.com>
Date: Mon, 7 Mar 2022 16:13:38 +0100
Subject: [PATCH] Use the correct name for installed gdk tests

---
 testsuite/gdk/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: gtk4-4.6.1+ds/testsuite/gdk/meson.build
===================================================================
--- gtk4-4.6.1+ds.orig/testsuite/gdk/meson.build
+++ gtk4-4.6.1+ds/testsuite/gdk/meson.build
@@ -83,7 +83,7 @@ if get_option('install-tests')
     test_cdata.set('testexecdir', testexecdir)
     test_cdata.set('test', test_name)
     configure_file(input: 'gdk.test.in',
-      output: '@0@.test'.format(t),
+      output: '@0@.test'.format(test_name),
       configuration: test_cdata,
       install: true,
       install_dir: testdatadir,
