From: Alessandro Astone <alessandro.astone@canonical.com>
Date: Mon, 16 Feb 2026 17:59:54 +0100
Subject: testsuite: Add missing test files to meson.build

Forwarded: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/9490
---
 testsuite/css/parser/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/testsuite/css/parser/meson.build b/testsuite/css/parser/meson.build
index bba2792..67731fb 100644
--- a/testsuite/css/parser/meson.build
+++ b/testsuite/css/parser/meson.build
@@ -363,6 +363,7 @@ test_data = [
   'filter-invalid9.errors',
   'filter-invalid9.ref.css',
   'filter-svg.css',
+  'filters.svg',
   'font-family.css',
   'font-family.ref.css',
   'font-features.css',
