From: =?utf-8?b?IllhbyBXZWkgKOmtj+mKmOW7tyki?= <yao.wei@canonical.com>
Date: Tue, 7 Oct 2025 14:44:15 +0800
Subject: gdk/x11: push dimensions if shadow size is updated

Forwarded: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/9017
Bug: https://gitlab.gnome.org/GNOME/gtk/issues/7805
Bug-Ubuntu: https://launchpad.net/bugs/1924689
---
 gdk/x11/gdksurface-x11.c | 23 ++++++++++++++++-------
 1 file changed, 16 insertions(+), 7 deletions(-)

diff --git a/gdk/x11/gdksurface-x11.c b/gdk/x11/gdksurface-x11.c
index 1f36535..038d444 100644
--- a/gdk/x11/gdksurface-x11.c
+++ b/gdk/x11/gdksurface-x11.c
@@ -206,7 +206,7 @@ gdk_x11_surface_update_size (GdkX11Surface *self,
   return TRUE;
 }
 
-static void
+static gboolean
 update_shadow_size (GdkSurface *surface,
                     int         shadow_left,
                     int         shadow_right,
@@ -222,7 +222,7 @@ update_shadow_size (GdkSurface *surface,
       impl->shadow_top == shadow_top &&
       impl->shadow_bottom == shadow_bottom &&
       impl->shadow_scale == impl->surface_scale)
-    return;
+    return FALSE;
 
   impl->shadow_left = shadow_left;
   impl->shadow_right = shadow_right;
@@ -242,6 +242,8 @@ update_shadow_size (GdkSurface *surface,
                    frame_extents, XA_CARDINAL,
                    32, PropModeReplace,
                    (guchar *) &data, 4);
+
+  return TRUE;
 }
 
 #define UPDATE_GEOMETRY TRUE
@@ -258,6 +260,7 @@ compute_toplevel_size (GdkSurface *surface,
   GdkMonitor *monitor;
   GdkToplevelSize size;
   int bounds_width, bounds_height;
+  gboolean shadow_updated = FALSE;
 
   monitor = gdk_display_get_monitor_at_surface (display, surface);
   if (monitor)
@@ -279,11 +282,11 @@ compute_toplevel_size (GdkSurface *surface,
 
   if (size.shadow.is_valid)
     {
-      update_shadow_size (surface,
-                          size.shadow.left,
-                          size.shadow.right,
-                          size.shadow.top,
-                          size.shadow.bottom);
+      shadow_updated = update_shadow_size (surface,
+                                           size.shadow.left,
+                                           size.shadow.right,
+                                           size.shadow.top,
+                                           size.shadow.bottom);
     }
 
   if (update_geometry)
@@ -335,6 +338,12 @@ compute_toplevel_size (GdkSurface *surface,
           impl->last_computed_width = size.width;
           impl->last_computed_height = size.height;
 
+          return TRUE;
+        }
+      if (shadow_updated)
+        {
+          *width = size.width;
+          *height = size.height;
           return TRUE;
         }
     }
