From: =?utf-8?b?IllhbyBXZWkgKOmtj+mKmOW7tyki?= <yao.wei@canonical.com>
Date: Thu, 9 Oct 2025 11:00:53 +0800
Subject: gdk: wrap scale checking code with X11 macros and checks

Forwarded: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/9017
Bug: https://gitlab.gnome.org/GNOME/gtk/issues/7805
Bug-Ubuntu: https://launchpad.net/bugs/1924689
---
 gdk/gdksurface.c        | 48 +++++++++++++++++++++++++++++++++++-------------
 gdk/gdksurfaceprivate.h |  8 ++++++--
 2 files changed, 41 insertions(+), 15 deletions(-)

diff --git a/gdk/gdksurface.c b/gdk/gdksurface.c
index 281823e..4bdf361 100644
--- a/gdk/gdksurface.c
+++ b/gdk/gdksurface.c
@@ -51,6 +51,10 @@
 #include <epoxy/egl.h>
 #endif
 
+#ifdef GDK_WINDOWING_X11
+#include "x11/gdkx11surface.h"
+#endif
+
 /**
  * GdkSurface:
  *
@@ -2072,20 +2076,29 @@ void
 gdk_surface_set_input_region (GdkSurface     *surface,
                               cairo_region_t *region)
 {
-  int current_scale;
+  gboolean region_unchanged;
 
   g_return_if_fail (GDK_IS_SURFACE (surface));
 
-  current_scale = gdk_surface_get_scale (surface);
-
   if (GDK_SURFACE_DESTROYED (surface))
     return;
 
-  if (cairo_region_equal (surface->input_region, region) &&
-      surface->input_region_scale == current_scale)
-    return;
+  region_unchanged = cairo_region_equal (surface->input_region, region);
+
+#ifdef GDK_WINDOWING_X11
+  if (GDK_IS_X11_SURFACE (surface))
+    {
+      int current_scale = gdk_surface_get_scale (surface);
+
+      region_unchanged = region_unchanged &&
+        (surface->input_region_scale == current_scale);
+
+      surface->input_region_scale = current_scale;
+    }
+#endif
 
-  surface->input_region_scale = current_scale;
+  if (region_unchanged)
+    return;
 
   if (surface->input_region)
     cairo_region_destroy (surface->input_region);
@@ -2629,18 +2642,27 @@ gdk_surface_set_opaque_region (GdkSurface      *surface,
                                cairo_region_t *region)
 {
   GdkSurfaceClass *class;
-  int current_scale;
+  gboolean region_unchanged;
 
   g_return_if_fail (GDK_IS_SURFACE (surface));
   g_return_if_fail (!GDK_SURFACE_DESTROYED (surface));
 
-  current_scale = gdk_surface_get_scale (surface);
+  region_unchanged = cairo_region_equal (surface->opaque_region, region);
 
-  if (cairo_region_equal (surface->opaque_region, region) &&
-      surface->opaque_region_scale == current_scale)
-    return;
+#ifdef GDK_WINDOWING_X11
+  if (GDK_IS_X11_SURFACE (surface))
+    {
+      int current_scale = gdk_surface_get_scale (surface);
+
+      region_unchanged = region_unchanged &&
+        (surface->opaque_region_scale == current_scale);
 
-  surface->opaque_region_scale = current_scale;
+      surface->opaque_region_scale = current_scale;
+    }
+#endif
+
+  if (region_unchanged)
+    return;
 
   g_clear_pointer (&surface->opaque_region, cairo_region_destroy);
 
diff --git a/gdk/gdksurfaceprivate.h b/gdk/gdksurfaceprivate.h
index 04ca620..15920a0 100644
--- a/gdk/gdksurfaceprivate.h
+++ b/gdk/gdksurfaceprivate.h
@@ -86,7 +86,9 @@ struct _GdkSurface
   GHashTable *device_cursor;
 
   cairo_region_t *input_region;
-  int input_region_scale;
+#ifdef GDK_WINDOWING_X11
+  int input_region_scale; /* only used for X11 backend */
+#endif
 
   GList *devices_inside;
 
@@ -96,7 +98,9 @@ struct _GdkSurface
   GdkDrawContext *paint_context;
 
   cairo_region_t *opaque_region;
-  int opaque_region_scale;
+#ifdef GDK_WINDOWING_X11
+  int opaque_region_scale; /* only used for X11 backend */
+#endif
 
   GdkSeat *current_shortcuts_inhibited_seat;
 
